internal-client/src/main/java/org/apache/uniffle/client/factory/ShuffleServerClientFactory.java

```diff
@@ -75,9 +80,30 @@ public synchronized ShuffleServerClient getShuffleServerClient(
     return serverToClients.get(shuffleServerInfo);
   }
 
+  public synchronized void cleanUselessShuffleServerClients(
```

Comment:
This method impl is not straightforward, I hope the impl like the following code:

```
public synchronized void closeClients(List<ShuffleServerInfo> candidates) {
    // ....
  }
```